### DISCUSSION TITLE
How use TableOfContent on Render.InteractiveAuto

### DISCUSSION THREAD

**Discussion by bao86** [microsoft/fluentui-blazor#1153] [Q&A] [answered]
Hi there,

How can I make the TableOfContent component work with the RenderMode.InteractiveAuto ?

Only works on InteractiveServer.

Thanks :)

**Comment by vnbaaij** [comment]
It can difinetely work with WebAssembly, as that is waht we are using on the demo site. 

Perhaps you need to add a `@rendermode` in the component you are including the TableOfContents in? Also, is the code in the Client project?

**Comment by bao86** [comment, marked as answer]
> It can difinetely work with WebAssembly, as that is waht we are using on the demo site.
> 
> Perhaps you need to add a `@rendermode` in the component you are including the TableOfContents in? Also, is the code in the Client project?
the TableOfContent component is include on MainLayout.razor and if I add @rendermode I got an error.

It works if I add @rendermode="InteractiveAuto" in <HeadOulet> and @rendermode="InteractiveServer" in <Routes > from App.razor
And the "counter" page (client side) is also working 👍 

If I add @rendermode="InteractiveServer" on both <HeadOulet> and <Routes>  , the TableOfContent is working but not the "counter" page.
